k8s_pod_autoscaler.tf

```diff
@@ -20,113 +20,5 @@ resource "kubernetes_horizontal_pod_autoscaler" "self" {
         name        = lookup(scale_target_ref.value, "name", null)
       }
     }
-    dynamic "metric" {
```

Comment:
Is there no replacement for this?

Reply:
No, based on the [documetation ](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/horizontal_pod_autoscaler) for `kuberentes_horizontal_pod_autoscaler`, the `metrics` configuration was a `v2beta2` feature and setting `metrics` would automatically make the resource use the v2beta2 API. So the only metric to base it on for the v1 version is `target_cpu_utilization_percentage`.

So this is available in `kubernetes_horizontal_pod_autoscaler_v2beta2` (see [documentation](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/horizontal_pod_autoscaler_v2beta2), can upgrade this resource to that if prefered?

Reply:
Yes, upgrade.

Reply:
Have updated it to use `kubernetes_horizontal_pod_autoscaler_v2`, and added the `metrics` block again.
This requires a minimum kubernetes version of v1.23 (which had an end of life date of 2023-02-28, so most clusters should be on a higher version).
Have updated documentation to address this